myapp/templates/layout.html

```diff
@@ -52,65 +49,33 @@
         function isNumeric(n) {
             return !isNaN(parseFloat(n)) && isFinite(n);
         }
-        // Validate the input when the admin sets up a lab
-        function Check_input(value_type,value_range,value_candidates){
-            var wrong_input = false;
-            var value_range_format = /([\d]+)-([\d]+)/;
-            var value_candidates_format = /([\w]+)((,([\w]+))+)/;
-            if (value_type!='Number' && value_type!='Text'){
-                alert('value_type must be either Number or Text!');
-                wrong_input = true;
-            }
-            if (value_range.length>0){
-                if (value_range.search(value_range_format)==-1){
-                    alert('value_range must be in the format 1-15');
-                    wrong_input = true;
-                }
-                if (value_type!='Number'){
-                    alert('value_range should only be filled when the value type is Number');
-                    wrong_input = true;
-                }
-            }
-            if (value_candidates.length>0){
-                if (value_candidates.search(value_candidates_format)==-1){
-                    alert('value_candidates must be in the format N,C');
-                    wrong_input = true;
-                }
-                if (value_type!='Text'){
-                    alert('value_candidates should only be filled when the value type is Text');
-                    wrong_input = true;
-                }
-            }
-            return wrong_input;
-        }
         // Validate the input when the student enters data
-        function Check_input_student(row_data,value_type,value_range,value_candidates){
-            var wrong_input = false;
-            var err_msg = '';
-            value_candidates
-            if (value_type=='Text' && value_candidates.length>0){
-                value_candidates_arr = value_candidates.split(',');
-                if (isInArray(row_data,value_candidates_arr)==false){
-                    wrong_input = true;
-                    err_msg = row_data+' is not in the value value candidates: '+value_candidates; 
+        function checkInputStudent(observation,valueType,valueRange,valueCandidates){
+            var wrongInput = false;
+            var errMsg = '';
+            valueCandidates
+            if (valueType=='Text' && valueCandidates.length>0){
+                valueCandidates_arr = valueCandidates.split(',');
```

Comment:
valueCandidates_arr should be valueCandidatesArr, and why not valueCandidatesArray while we're at it? Avoiding abbreviations where possible is a good thing.